packages/auth-construct/src/construct.ts

```diff
@@ -900,132 +902,209 @@ export class AmplifyAuth
       Stack.of(this)
     )
   ): void => {
+    // these properties cannot be overwritten
     const output: AuthOutput['payload'] = {
       userPoolId: this.resources.userPool.userPoolId,
       webClientId: this.resources.userPoolClient.userPoolClientId,
       identityPoolId: this.resources.cfnResources.cfnIdentityPool.ref,
       authRegion: Stack.of(this).region,
-      allowUnauthenticatedIdentities:
+    };
+
+    // the properties below this line can be overwritten, so they are exposed via cdk LAZY
```

Comment:
Is it not possible to extract the entire `cfnUserPool` once instead of for each property?

Reply:
yes, just updated thanks

Reply:
doing this would produce an IResolvable for the cfnUserPool , which is not something we can access properties of (if we were trying to build the output from the IResolvable)